data-protocols/dsp/dsp-2025/dsp-http-api-configuration-2025/src/main/java/org/eclipse/edc/protocol/dsp/http/api/configuration/v2025/DspApiConfigurationV2025Extension.java

```diff
@@ -0,0 +1,152 @@
+import org.eclipse.edc.transform.transformer.edc.to.JsonObjectToQuerySpecTransformer;
+import org.eclipse.edc.transform.transformer.edc.to.JsonValueToGenericTypeTransformer;
+import org.eclipse.edc.web.spi.configuration.ApiContext;
+import org.eclipse.edc.web.spi.configuration.PortMapping;
+
+import java.util.Map;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static org.eclipse.edc.jsonld.spi.Namespaces.DSPACE_CONTEXT_2025_1;
+import static org.eclipse.edc.jsonld.spi.Namespaces.EDC_DSPACE_CONTEXT;
+import static org.eclipse.edc.protocol.dsp.http.spi.types.HttpMessageProtocol.DATASPACE_PROTOCOL_HTTP_V_2025_1;
+import static org.eclipse.edc.protocol.dsp.spi.type.DspConstants.DSP_NAMESPACE_V_2025_1;
+import static org.eclipse.edc.protocol.dsp.spi.type.DspConstants.DSP_SCOPE_V_2025_1;
+import static org.eclipse.edc.protocol.dsp.spi.type.DspConstants.DSP_TRANSFORMER_CONTEXT_V_2025_1;
+import static org.eclipse.edc.protocol.dsp.spi.version.DspVersions.V_2025_1_PATH;
+import static org.eclipse.edc.spi.constants.CoreConstants.JSON_LD;
+
+/**
+ * Configure 'protocol' api context.
+ */
+@Extension(value = DspApiConfigurationV2025Extension.NAME)
+public class DspApiConfigurationV2025Extension implements ServiceExtension {
+
+    public static final String NAME = "Dataspace Protocol 2025/1 API Configuration Extension";
+
+    static final String DEFAULT_PROTOCOL_PATH = "/api/protocol";
+    static final int DEFAULT_PROTOCOL_PORT = 8282;
+
+    @Setting(description = "Configures endpoint for reaching the Protocol API in the form \"<hostname:protocol.port/protocol.path>\"", key = "edc.dsp.callback.address", required = false)
+    private String callbackAddress;
+
+    @Configuration
+    private DspApiConfiguration apiConfiguration;
+
+    @Inject
+    private TypeManager typeManager;
+    @Inject
+    private JsonLd jsonLd;
+    @Inject
+    private TypeTransformerRegistry transformerRegistry;
+    @Inject
+    private ParticipantIdMapper participantIdMapper;
+    @Inject
+    private RemoteMessageDispatcherRegistry dispatcherRegistry;
+    @Inject
+    private DspHttpRemoteMessageDispatcher dispatcher;
+
+    @Inject
+    private ProtocolWebhookRegistry protocolWebhookRegistry;
+
+    @Inject
+    private Hostname hostname;
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var portMapping = new PortMapping(ApiContext.PROTOCOL, apiConfiguration.port(), apiConfiguration.path());
+
+        // registers ns for DSP 2025/1 scope
+        registerNamespaces();
+        registerTransformers();
+        dispatcherRegistry.register(DATASPACE_PROTOCOL_HTTP_V_2025_1, dispatcher);
+
+        var dspWebhookAddress = ofNullable(callbackAddress).orElseGet(() -> format("http://%s:%s%s", hostname.get(), portMapping.port(), portMapping.path()));
```

Comment:
maybe the "base" DSP webhook address could be registered by the DSP api configuration module and be used eventually by other extensions like this.
Or it could be also registered on the `ProtocolWebhookRegistry` as `baseWebhook` or something like this

Reply:
Yeah would have to figure it out how to expose the base webhook address. I would raise an issue about this to be tackle in a separated PR.